imageio/imageio-tiff/src/main/java/com/twelvemonkeys/imageio/plugins/tiff/CCITTFaxDecoderStream.java

```diff
@@ -366,17 +382,7 @@ private int decodeRun(final Tree tree) throws IOException {
     }
 
     private void resetBuffer() throws IOException {
-        for (int i = 0; i < decodedRow.length; i++) {
-            decodedRow[i] = 0;
-        }
-
-        while (true) {
-            if (bufferPos == -1) {
-                return;
-            }
-
-            readBit();
-        }
+        bufferPos = -1;
```

Comment:
This code effectively never needed:
readBit() also just counts up bufferPos und returned the current bool.
Reseting decodedRow also is done in the filling loop starting on line 311.
I'm sure the present unit tests would have flagged it if it changed behaviour.
I'm just now realized this code is redundant.